tests/ludwig/schema/test_config_object.py

```diff
@@ -0,0 +1,373 @@
+
+    assert config_object.trainer.epochs == 50
+    assert config_object.trainer.batch_size == "auto"
+
+    assert config_object.trainer.optimizer.type == "adam"
+    assert config_object.trainer.optimizer.beta1 == 0.8
+    assert config_object.trainer.optimizer.beta2 == 0.999
+    assert config_object.trainer.optimizer.epsilon == 5e-09
+
+
+def test_config_object_defaults():
+    config = {
+        "input_features": [
+            {"name": "number_feature", "type": "number"},
+            {
+                "name": "text_feature_1",
+                "type": "text",
+                "encoder": {
+                    "type": "rnn",
+                    "activation": "sigmoid",
+                },
+            },
+            {
+                "name": "text_feature_2",
+                "type": "text",
+            },
+        ],
+        "output_features": [
+            {
+                "name": "number_output_feature",
+                "type": "number",
+            },
+        ],
+        "defaults": {
+            "number": {"preprocessing": {"missing_value_strategy": "drop_row"}, "encoder": {"type": "dense"}},
+            "text": {
+                "preprocessing": {
+                    "missing_value_strategy": "drop_row",
+                },
+                "encoder": {
+                    "type": "stacked_parallel_cnn",
+                    "activation": "tanh",
+                },
+            },
+        },
+    }
+
+    config_object = ModelConfig.from_dict(config)
+    assert config_object.input_features.number_feature.preprocessing.missing_value_strategy == "drop_row"
+    assert config_object.input_features.number_feature.encoder.type == "dense"
+
+    assert config_object.input_features.text_feature_1.encoder.type == "rnn"
+    assert config_object.input_features.text_feature_1.encoder.activation == "sigmoid"
+    assert config_object.input_features.text_feature_1.preprocessing.missing_value_strategy == "drop_row"
+
+    assert config_object.input_features.text_feature_2.encoder.type == "stacked_parallel_cnn"
+    assert config_object.input_features.text_feature_2.encoder.activation == "tanh"
+    assert config_object.input_features.text_feature_2.preprocessing.missing_value_strategy == "drop_row"
+
+
+def test_config_object_to_config_dict():
+    config = {
+        "input_features": [
+            {"name": "number_feature", "type": "number"},
+        ],
+        "output_features": [
+            {
+                "name": "number_output_feature",
+                "type": "number",
+            },
+        ],
+    }
+
+    config_object = ModelConfig.from_dict(config)
+    config_dict = config_object.to_dict()
+
+    assert INPUT_FEATURES in config_dict
+    assert OUTPUT_FEATURES in config_dict
+    assert PREPROCESSING in config_dict
+    assert TRAINER in config_dict
+    assert COMBINER in config_dict
+    assert DEFAULTS in config_dict
+    assert len(config_dict[DEFAULTS]) == 13
+    assert HYPEROPT in config_dict
```

Comment:
This could be difficult to maintain when we add new config sections, or new types.

I would more simply do, as the scope of this test should be focused to just test that the to_dict mechanism doesn't introduce any inconsistencies (I think):

```python
assert config_dict[INPUT_FEATURES] == {...}
assert config_dict[OUTPUT_FEATURES] == {...}
```

Reply:
This is a good catch, changed up a test a little to capture the scope better!